asserts/asserts.go

```diff
@@ -28,6 +28,12 @@ import (
 	"strconv"
 )
 
+type typeFlags int
+
+const (
+	freestanding typeFlags = iota + 1
```

Comment:
Can we please use hasAuthority here instead, and kill the term "freestanding" completely while we have the chance? :)